src/s2s/pipeline.cpp

```
@@ -0,0 +1,1756 @@
+    double cum = 0;
+    int keep = V;
+    for (int i = 0; i < V; i++) {
+        cum += p[i] / denom;
+        if (cum > cfg_.top_p) {
+            keep = i + 1;
+            break;
+        }  // shift-by-one: keep top-1+
+    }
+    double sub = 0;
+    for (int i = 0; i < keep; i++) sub += p[i];
+    double r = static_cast<double>(rand()) / RAND_MAX * sub;
```

Clang (14.0.6)

[warning] variable name 'r' is too short, expected at least 3 characters

(readability-identifier-length)